packages/zosfiles/src/methods/list/List.ts

```diff
@@ -77,35 +74,23 @@ export class List {
 
             this.log.debug(`Endpoint: ${endpoint}`);
 
-            let data = await ZosmfRestClient.getExpectString(session, endpoint, reqHeaders);
+            const data = await ZosmfRestClient.getExpectString(session, endpoint, reqHeaders);
             let response: any;
             try {
                 response = JSONUtils.parse(data);
-            } catch {
-                const regex = /"member":\s*"/g;
-                let match;
-                const matches = [];
-
-                while ((match = regex.exec(data)) !== null) {
-                    matches.push({
-                        matchString: match[0],
-                        startIndex: match.index
-                    });
-                }
-
-                // Iterate through matches in reverse order
-                for (let i = matches.length - 1; i >= 0; i--) {
-                    const match = matches[i];
-                    const memberStartIdx = match.startIndex + match.matchString.length;
-                    const memberNameEndIdx = data.indexOf('"', memberStartIdx + 1); // Find the end of the member name
-                    if (memberNameEndIdx !== -1) {
-                        const memberName = data.substring(memberStartIdx, memberNameEndIdx);
-                        const escapedMemberName = memberName.replace(/(["\\])/g, `\\$1`).replace(this.CONTROL_CHAR_REGEX, "\\ufffd");
-                        data = data.substring(0, memberStartIdx) + escapedMemberName + data.substring(memberNameEndIdx);
-                    }
+            } catch (err) {
+                const match = /in JSON at position (\d+)/.exec(err.message);
+                if (match != null) {
+                    // Remove invalid member names from end of list and try to parse again
+                    const lineNum = data.slice(0, parseInt(match[1])).split("\n").length - 1;
+                    const lines = data.trim().split("\n");
+                    lines[lineNum - 1] = lines[lineNum - 1].replace(/,$/, "");
+                    lines.splice(lineNum, lines.length - lineNum - 1);
+                    response = JSONUtils.parse(lines.join("\n"));
+                    response.items.push({ "member": `… ${response.returnedRows - response.items.length} more members` });
```

Comment:
@anaxceron What do we want the message to say to make it clear there was an error loading these members? Something like "N invalid members" or "N members with errors"?

The output will look something like this:

```
$ zowe files list all-members IBMUSER.PDS
MEMBER1
MEMBER2
...3 invalid members
```

Reply:
Just replying to say that we discussed this in a Google Chat and agreed on the message. Thank you!